packages/devtools_app/lib/src/survey.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:convert';
+
+import 'dart:io' if (dart.library.html) 'dart:html';
+
+import 'package:flutter/widgets.dart';
+import 'package:http/http.dart';
+
+import 'config_specific/logger/logger.dart';
+import 'config_specific/server/server.dart' as server;
+import 'notifications.dart';
+import 'utils.dart';
+
+class SurveyService {
+  static const noThanksLabel = 'NO THANKS';
+
+  static const takeSurveyLabel = 'TAKE SURVEY';
+
+  static const maxShowSurveyCount = 5;
+
+  static const metadataUrl =
+      'https://flutter.dev/f/dart-devtools-survey-metadata.json';
+
+  /// Duration for which we should show the survey notification.
+  ///
+  /// We use a very long time here to give the appearance of a persistent
+  /// notification. The user will need to interact with the prompt to dismiss
+  /// it.
+  static const notificationDuration = Duration(days: 1);
+
+  /// The amount of time (one day) to wait between checks to the file at
+  /// [metadataUrl].
+  static const checkInterval = Duration(days: 1);
```

Comment:
why do we bother with this? just loading the file and relying on browser caching should be fine as long as we don't block the devtools UI waiting for the json file to be loaded.